### PR TITLE
feat: fiat ramp back button

### LINKED ISSUE #2273: No back button in the fiat ramp modal
### Overview

When selecting the $Buy/Sell Crypto button in app.shapeshift, there is no back back if the user changed their mind on the ramp of choice. They can only exit the modal and start from the beginning again .

### References and additional details

1) Log into app.shapeshift.com 
2) Connect to wallet of choice
3) Select the $Buy/Sell Crypto button
4) Once you choose a ramp, notice there is no back button. If you change your mind on the fiat ramp you chose, you need to X out of the modal and start again . 

### Acceptance Criteria

A back button would allow you to change ramps without exiting all together and having to start again 

### Need By Date

_No response_

### Screenshots/Mockups

_No response_

### Ownership

- [X] If my bounty needs engineering or needs product I have added the respective labels on the right
- [X] As the sponsor of this bounty I will review the changes in a preview environment (ops/product) or review the PR (engineering)

### Estimated effort

_No response_

### Sponsor / Stakeholder

_No response_

### Bounty Hunters

- [Join our discord](https://discord.gg/shapeshift)
- Include an expected timeline for you to complete work in the work plan when you apply for this bounty!
- Please refer to [this link](https://shapeshiftdao.gitbook.io/getting-started/how-to-contribute/bounties#resources-for-new-contributors-and-bounty-hunters) for some basic info
- Please ***do not*** start work on this issue until you are approved in Gitcoin.

### PR DESCRIPTION
**Pull request by jmurphy5613**
## Description

I added a back button to the fiat ramp modal. This is connected to [this issue](https://github.com/shapeshift/web/issues/2273).

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes #2273 

## Risk

Small, ui changes to only the fiat ramp modal

## Testing

- Open the buy/sell crypto button on the dashboard
- Click on any of the ramps
- click the back button at the top left

## Screenshots (if applicable)
Screen Shot 2022-07-30 at 4 11 25 AM

Screen Shot 2022-07-30 at 4 11 14 AM

### CODE REVIEW

**Review by gomesalexandre — CHANGES_REQUESTED** [review]
Hi @jmurphy5613 👋🏽 thank you for the first contribution!

Could you please get lint autofix set up on your editor or run `yarn lint` / `yarn local-ci` and fix the lint errors accordingly?

WRT the functionality of this, it does work as intended when you're on the "Asset to buy / Asset to sell" route but when you're on a "Select an asset to buy" / "Select an asset to sell" route, there are now two back buttons, and the modal header is collapsed:

This diff:
image

develop:

image

The larger one (the newly introduced one) routes back to the fiats ramp selection, while the smaller one routes back to the buy/sell screen.

We should keep only one (preferably the larger one, which is actually a more consistent one throughout the app) and make it superseed the behavior or the "smaller" one e.g route back to the buy/sell section if on the asset selection route, or route back to the fiat ramps selection if on the buy/sell section

**Comment by jmurphy5613** [comment]
Thanks for your feedback. How can I get access to the current location or url of the Manager components within the FiatRamps component without using something like the context api? The MemoryRouter isn't tracking the history of the nested routes.

**Comment by gomesalexandre** [comment]
> Thanks for your feedback. How can I get access to the current location or url of the Manager components within the FiatRamps component without using something like the context api? The MemoryRouter isn't tracking the history of the nested routes.

I just gave it a try locally and yeah, our usage of `<MemoryRouter />` will make this impossible without revamping the routes logic. That would be out of the scope of this bounty.

@shapeshift/product What do we think about keeping only the newly added outer button, but losing the functionality of going back from "Select an asset to buy" / "Select an asset to sell" to "Asset to buy" / "Asset to sell" (it will route back to the ramps list) until the core team tackles the architectural change necessary to make it work properly?

File: src/components/Modals/FiatRamps/FiatRamps.tsx
Comment (gomesalexandre) [inline src/components/Modals/FiatRamps/FiatRamps.tsx:37]: You can remove that wrapping arrow function to make it one less function in the stack trace

```suggestion
          onClick={history.goBack}
```

**Comment by gomesalexandre** [comment]
Checked out locally and CI is passing. @shapeshift/product WRT https://github.com/shapeshift/web/pull/2284#issuecomment-1200280054 please use the Fleek build from this PR:

https://bafybeiffsd5b6o3weowpk23yzqpq2o4m726jcxrzh77c4quksjjw3id2ye.on.fleek.co/

**Comment by decentralized-j** [comment]
> > Thanks for your feedback. How can I get access to the current location or url of the Manager components within the FiatRamps component without using something like the context api? The MemoryRouter isn't tracking the history of the nested routes.
> 
> I just gave it a try locally and yeah, our usage of `<MemoryRouter />` will make this impossible without revamping the routes logic. That would be out of the scope of this bounty.
> 
> @shapeshift/product What do we think about keeping only the newly added outer button, but losing the functionality of going back from "Select an asset to buy" / "Select an asset to sell" to "Asset to buy" / "Asset to sell" (it will now route back to the ramps list) until the core team tackles the architectural change necessary to make it work properly?
> 
> Or alternatively, keep the two, which might look weird from UI standpoint.

I think this is a fine approach to take until the bigger lift can be conducted. At the end of the day I think it's a win for users to be able to back out to the provider list rather than having to start the entire process over again and reopen the modal. We don't have the data to back it up but I'd guess our conversion rate would be higher in this instance.

I ran through the Fleek link and did my best to break it, this functions exactly as expected in my testing across all 3 providers. Curious if ops can break it somehow but if they can't then this looks ready to roll to me! thank you @gomesalexandre @jmurphy5613

File: src/components/Modals/FiatRamps/FiatRamps.tsx
Comment (gomesalexandre) [reply src/components/Modals/FiatRamps/FiatRamps.tsx:17]: Got some eqeqeq tweak spinned up so we error on this vs. simply emitting a warning currently: https://github.com/shapeshift/web/pull/2291

**Review by gomesalexandre — CHANGES_REQUESTED** [review]

**Review by gomesalexandre — APPROVED** [review]
From a bounty perspective, this is it! Well done @jmurphy5613 👍🏽 

Keep in mind you'll need another stamp here before this gets merged and the bounty is completed

**Review by 0xdef1cafe — APPROVED** [review]

**Pull request merged**
